aiounifi/devices.py

```diff
@@ -345,3 +384,77 @@ def __getitem__(self, obj_id: int) -> Port:
     def __iter__(self) -> Iterator[Union[int, str]]:
         """Iterate over ports."""
         return iter(self.ports)
+
+
+class Outlet:
+    """Represents an outlet."""
+
+    def __init__(self, raw: dict) -> None:
+        """Initialize outlet."""
+        self.raw = raw
+
+    @property
+    def name(self) -> str:
+        """Name of outlet"""
+        return self.raw.get("name", "")
+
+    @property
+    def index(self) -> int:
+        """Outlet index."""
+        return self.raw.get("index")
+
+    @property
+    def has_relay(self) -> bool:
+        """Is the outlet controllable"""
+        return self.raw.get("has_relay", False)
+
+    @property
+    def has_metering(self) -> Optional[bool]:
+        """Is metering supported."""
+        return self.raw.get("has_metering")
+
+    @property
+    def relay_state(self) -> Optional[bool]:
+        """Is outlet power on."""
+        return self.raw.get("relay_state")
+
+
+class Outlets:
+    """Represents outlets on a device."""
+
+    def __init__(self, raw: List[dict]) -> None:
+        """Initialize port manager."""
+        self.outlets: Dict[Union[int, str], Port] = {}
+        for raw_outlet in raw:
+            outlet = Outlet(raw_outlet)
+
+            if (index := outlet.index) is not None:
```

Comment:
Can index be None? It would be good if it could be expected to exist

Reply:
From the devices I've seen so far, index always exists and in an integer.  Will fix.